packages/api/src/payments/paymentProvider.ts

```diff
@@ -129,31 +137,97 @@ export abstract class BasePaymentProvider implements PaymentProvider {
 
     if (!updatedPayment) throw new Error('Error while updating Payment')
 
+    // get invoice and subscription joins out of the payment
```

Comment:
the whole logic with activating the user is handled in https://github.com/wepublish/wepublish/blob/f/WPC-309-multiple-subscriptions/packages/api/src/events.ts#L176

Reply:
Will have a look at it.

Reply:
Why I introduced this code: If an user calls the `registerAndReceivePayment()` mutation and then the api receives the webhook eg. from Payrexx the following line would always evaluate false cause paidAt is obviously null: https://github.com/wepublish/wepublish/blob/eea4da1b7164318659255cbda138fa1b88cdd0c0/packages/api/src/events.ts#L178

How I tested: I called the registerAndReceivePayment() mutation and then fired the webhook from Payrexx to the api. The invoice remained unpaid.

How would you solve this problem though?